projects/plugins/super-cache/inc/delete-cache-button.php

```diff
@@ -86,6 +86,17 @@ function wpsc_admin_bar_delete_cache_ajax() {
 		if ( defined( 'WPSCDELETEERROR' ) ) {
 			return json_decode( constant( 'WPSCDELETEERROR' ) );
 		} else {
+			$req_path = isset( $_POST['path'] ) ? sanitize_text_field( stripslashes( $_POST['path'] ) ) : '';
```

Comment:
PHP Code Sniffer found some (longstanding) issues here that we should look at. The first is the nonce check. Can you add this code right at the top of the function?

```
	if ( ! isset( $_POST['path'] ) ) {
		$_POST['path'] = '/';
	}

	if ( ! isset( $_POST['nonce'] ) ) {
		wp_send_json_error( 'missing nonce.', 401 );
	}

	if ( ! wp_verify_nonce( $_POST['nonce'], 'delete-cache-' . wp_unslash( $_POST['path'] ) . '_1' ) ) { // "1" for admin
		wp_send_json_error( 'Invalid nonce.', 401 );
	}
```
That takes care of the nonce check.

Can you change `stripslashes` on line 89 to `wp_unslash`? That will satisfy the other PHPCS check.

Reply:
One more change to support returning an error. The file delete-cache-button.js needs the following code around line 24, just before the "complete:" section
```
error: function ( msg ) {
    console.log( 'Error deleting cache' );
},
```

Make sure it aligns with the rest of the code, please.

Sorry about making this patch more complicated. Jetpack has strict code checks, and they notice things that are missing.